overlord/state/warning.go

```diff
@@ -262,8 +274,10 @@ func (s *State) AllWarnings() []*Warning {
 
 // OkayWarnings marks warnings that were showable at the given time as shown.
 func (s *State) OkayWarnings(t time.Time) int {
-	t = t.UTC()
 	s.writing()
+	s.warningsMu.Lock()
+	defer s.warningsMu.Unlock()
+	t = t.UTC()
```

Comment:
prob not time consuming but conversion to UTC could be done before taking the lock